pallets/multisig-verifier/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+    impl Default for GenesisConfig {
+        fn default() -> Self {
+            Self {
+                network_id: GenericNetworkId::Sub(SubNetworkId::Mainnet),
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {
+            ThisNetworkId::<T>::put(self.network_id);
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(0)]
+        pub fn initialize(
+            origin: OriginFor<T>,
+            network_id: GenericNetworkId,
+            keys: Vec<ecdsa::Public>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            ensure!(keys.len() > 0, Error::<T>::InvalidInitParams);
+
+            let btree_keys = keys.into_iter().collect();
+            PeerKeys::<T>::set(network_id, Some(btree_keys));
+            Self::deposit_event(Event::NetworkInitialized(network_id));
+            Ok(().into())
+        }
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(0)]
+        pub fn add_peer(
+            origin: OriginFor<T>,
+            network_id: GenericNetworkId,
+            key: ecdsa::Public,
+        ) -> DispatchResultWithPostInfo {
+            let output = T::CallOrigin::ensure_origin(origin)?;
+            frame_support::log::info!(
+                "Call add_peer {:?} by {:?}",
+                key,
+                output
+            );
+            PeerKeys::<T>::try_mutate(network_id,|x| -> DispatchResult {
+                let Some(keys) = x else {
+                    fail!(Error::<T>::NetworkNotInitialized)
+                };
+                keys.insert(key);
+                Ok(())
+            })?;
+            Self::deposit_event(Event::PeerAdded(key));
+            Ok(().into())
+        }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight(0)]
+        pub fn remove_peer(
+            origin: OriginFor<T>,
+            network_id: GenericNetworkId,
+            key: ecdsa::Public,
+        ) -> DispatchResultWithPostInfo {
+            let output = T::CallOrigin::ensure_origin(origin)?;
+            frame_support::log::info!(
+                "Call remove_peer {:?} by {:?}",
+                key,
+                output
+            );
+            PeerKeys::<T>::try_mutate(network_id,|x| -> DispatchResult {
+                let Some(keys) = x else {
+                    fail!(Error::<T>::NetworkNotInitialized)
+                };
+                ensure!(keys.remove(&key), {
+                    Self::deposit_event(Event::NoSuchPeer(key));
+                    Error::<T>::NoSuchPeer
+                });
+                Ok(())
+            })?;
+            Self::deposit_event(Event::PeerRemoved(key));
+            Ok(().into())
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        pub fn verify_signatures(
+            network_id: GenericNetworkId,
+            hash: &H256,
```

Comment:
Copy trait implemented for H256
```suggestion
            hash: H256,
```